src/Node/HTTP.purs

```diff
@@ -51,6 +77,194 @@ foreign import data HTTP :: Effect
 -- | Create a HTTP server, given a function to be executed when a request is received.
 foreign import createServer :: forall eff. (Request -> Response -> Eff (http :: HTTP | eff) Unit) -> Eff (http :: HTTP | eff) Server
 
+-- | The type of HTTPS server options
+data SSLOptions
+
+-- | Abort the connection if the SSL/TLS handshake does not finish in the
+-- | specified number of milliseconds. Defaults to 120 seconds. A
+-- | 'tlsClientError' is emitted on the tls.Server object whenever a handshake
+-- | times out.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+handshakeTimeout :: Option SSLOptions Int
+handshakeTimeout = opt "handshakeTimeout"
+
+-- | If true the server will request a certificate from clients that connect and
+-- | attempt to verify that certificate. Defaults to false.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+requestCert :: Option SSLOptions Boolean
+requestCert = opt "requestCert"
+
+-- | If not false the server will reject any connection which is not authorized
+-- | with the list of supplied CAs. This option only has an effect if
+-- | requestCert is true. Defaults to true.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+rejectUnauthorized :: Option SSLOptions Boolean
+rejectUnauthorized = opt "rejectUnauthorized"
+
+-- | An array of strings, Buffers or Uint8Arrays, or a single Buffer or
+-- | Uint8Array containing supported NPN protocols. Buffers should have the
+-- | format [len][name][len][name]... e.g. 0x05hello0x05world, where the first
+-- | byte is the length of the next protocol name. Passing an array is usually
+-- | much simpler, e.g. ['hello', 'world']. (Protocols should be ordered by
+-- | their priority.)
+-- | The type variable t should be a string[], Buffer[], Uint8Array[], Buffer,
+-- | or Uint8Array.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+npnProtocols :: forall t. Option SSLOptions t
```

Comment:
Does anyone know a better way to use the `Option` type to bind to Node APIs that are built to accept multiple different types?  For now I just accepted any type and documented the list of types that the Node API expects... which I guess is OK, but it would be nice to actually type it correctly